### PR TITLE
Extended multi-endpoint E2E direct method tests.

### PR DESCRIPTION
**Pull request by karok2m**
Changes:
* In `LegacyJobOrchestrator` :
  * Added call to `TriggerAgentConfigUpdate()` in the end of `AddOrUpdateEndpointsAsync()` method to quickly notify workers of configuration changes.
  * Added logging of execution time to `GetConfiguredEndpointsAsync()` method.
* Some refactoring of direct method tests:
  * Added `DirectMethodTestBase` which (1) does the initialization of required objects for the test classes which was previously done similarly in the test constructors and (2) adds a wrapper method for calling direct methods.
  * Added `ModuleDeploymentConfiguration` base class for modern and legacy OPC Publisher deployments. The class now holds the name of the module that is deployed. Now we can use the value for the module name from the class itself instead of hardcoding it in tests.
* Extended multi-endpoint test to check for all nodes. Now the two endpoints are monitoring different nodes. So with `TestEventProcessor` we can see if values from both are coming. This is required as `TestEventProcessor` is not yet able to differentiate source endpoints of nodes and report accordingly.
* Changed cleanup to update published nodes JSON files of both versions of OPC Publisher. Added `CleanPublishedNodesJsonFilesAsync()` method to do that.
* Fixed expected node generation.
* Fixed multi-endpoint tests for legacy OPC Publisher.

### CODE REVIEW

File: e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs [hunk 1]
```diff
@@ -212,32 +212,54 @@ internal static partial class TestHelper {
...
                 }
             }
         }
 
+        /// <summary>
+        /// Clean published nodes JSON files for both legacy (2.5) and current (2.8) versions.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public static async Task CleanPublishedNodesJsonFiles(IIoTPlatformTestContext context) {
```
Comment (karok2m) [inline e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs]: Method name should have `Async` in the end.

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs [hunk 2]
```diff
@@ -64,156 +42,226 @@ IIoTMultipleNodesTestContext context
...
+                foreach (var consumedNodes in _context.ConsumedOpcUaNodes) {
+                    foreach (var opcNode in consumedNodes.Value.OpcNodes) {
+                        expectedNodes.Add(opcNode.Id);
+                    }
+                }
+
                 foreach (dynamic property in publishingMonitoringResultJson.valueChangesByNodeId) {
                     var propertyName = (string)property.Name;
                     var nodeId = propertyName.Split('#').Last();
+
```
Comment (dacolgit) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs]: remove empty line

Comment (dacolgit) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs]: remove empty line

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs [hunk 3]
```diff
@@ -247,126 +295,199 @@ IIoTMultipleNodesTestContext context
...
```
Comment (dacolgit) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs]: remove empty line

File: e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs (hunk 1 above)
Comment (karok2m) [reply e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs]: Added.

**Review by dacolgit — APPROVED** [review]

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs (hunk 3 above)
Comment (karok2m) [reply e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs]: Removed.

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs (hunk 2 above)
Comment (karok2m) [reply e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs]: Removed.

Comment (karok2m) [reply e2e-tests/IIoTPlatform-E2E-Tests/Standalone/C_PublishMultipleEndpointsStandaloneDirectMethodTestTheory.cs]: Done.

**Review by cristipogacean — APPROVED** [review]
:shipit:

**Pull request merged**